src/main/java/com/socobo/security/service/RegistrationImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.socobo.security.service;
+
+import com.socobo.security.exception.RegistrationException;
+import com.socobo.security.model.Role;
+import com.socobo.security.model.User;
+import com.socobo.security.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+/**
+ * Created by patrick on 05.11.16.
+ */
+@Service
+public class RegistrationImpl implements Registration{
+
+    private UserRepository userRepository;
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    public RegistrationImpl(UserRepository userRepository, PasswordEncoder passwordEncoder) {
+        this.userRepository = userRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Override
+    public User register(User user) {
+        verifyUserDoesNotExist(user);
```

Comment:
Ok, then I will rename the method. I think its better than to pollute the register method with an additional if statement